### PR TITLE
439 Termination fix hanging bug exposed in send chain

### PR DESCRIPTION
**Pull request by lifflander**

### CODE REVIEW

**Comment by lifflander** [comment]
We have a failure in `test_reduce_collection`; I don't think it's related, but looking into it.

**Comment by lifflander** [comment]
@PhilMiller @pnstickne @nlslatt Can one of you review this change so I can get it merged? It is blocking several other pull requests.

File: tests/unit/termination/test_term_dep_send_chain.cc [hunk 1]
```diff
@@ -277,7 +277,6 @@ struct MyCol : vt::Collection<MyCol,vt::Index2D> {
       migrating = false;
     }
     s | op6_counter_;
-    op6_msgs_ = {};
```
Comment (nlslatt) [inline tests/unit/termination/test_term_dep_send_chain.cc]: If you're going to ignore this member when serializing, please document why.

**Comment by nlslatt** [comment]
I still see a termination hang in `vt:*/TestTermDepSendChain.test_term_dep_send_chain/*_proc_4`.  Is that expected to be addressed in a different PR?

I also see a timeout in `vt:*/TestTermDepSendChain.test_term_dep_send_chain/*_proc_16`.  For some reason the timeout is shorter than for the other processor counts.  We should fix that.

File: src/vt/serialization/messaging/serialized_messenger.impl.h [hunk 2]
```diff
@@ -136,8 +144,19 @@ template <typename UserMsgT>
...
       runnable::Runnable<UserMsgT>::run(handler, nullptr, msg.get(), node);
       action();
+
+      if (is_valid_epoch) {
+        theMsg()->popEpoch(epoch);
+      }
+
+      if (is_valid_epoch) {
+        theTerm()->consume(epoch);
+      }
```
[the remark this answers is not in the document]
Comment (lifflander) [reply src/vt/serialization/messaging/serialized_messenger.impl.h]: Fixed.

File: tests/unit/termination/test_term_dep_send_chain.cc (hunk 1 above)
Comment (lifflander) [reply tests/unit/termination/test_term_dep_send_chain.cc]: Added comment and assert about the state when sizing/packing.

**Comment by nlslatt** [comment]
> I still see a termination hang in `vt:*/TestTermDepSendChain.test_term_dep_send_chain/*_proc_4`. Is that expected to be addressed in a different PR?
> 
> I also see a timeout in `vt:*/TestTermDepSendChain.test_term_dep_send_chain/*_proc_16`. For some reason the timeout is shorter than for the other processor counts. We should fix that.

I may have been oversubscribing a node this morning, causing the termination hang logic to kick in inappropriately.  I can't currently reproduce the hang.

**Comment by nlslatt** [comment]
> > I still see a termination hang in `vt:*/TestTermDepSendChain.test_term_dep_send_chain/*_proc_4`. Is that expected to be addressed in a different PR?
> > I also see a timeout in `vt:*/TestTermDepSendChain.test_term_dep_send_chain/*_proc_16`. For some reason the timeout is shorter than for the other processor counts. We should fix that.
> 
> I may have been oversubscribing a node this morning, causing the termination hang logic to kick in inappropriately. I can't currently reproduce the hang.

Here is a portion of my ctest log, if it helps you decide if there's a real problem. 
[LastTest.td.log](https://github.com/DARMA-tasking/vt/files/3709708/LastTest.td.log)

**Comment by nlslatt** [comment]
> > > I still see a termination hang in `vt:*/TestTermDepSendChain.test_term_dep_send_chain/*_proc_4`. Is that expected to be addressed in a different PR?
> > > I also see a timeout in `vt:*/TestTermDepSendChain.test_term_dep_send_chain/*_proc_16`. For some reason the timeout is shorter than for the other processor counts. We should fix that.
> > 
> > 
> > I may have been oversubscribing a node this morning, causing the termination hang logic to kick in inappropriately. I can't currently reproduce the hang.
> 
> Here is a portion of my ctest log, if it helps you decide if there's a real problem.
> [LastTest.td.log](https://github.com/DARMA-tasking/vt/files/3709708/LastTest.td.log)

I've been able to reproduce this by only allocating one physical node on Kahuna but running the 4-proc MPI job.  When I run with 4 physical nodes, I don't get messages about termination detection hangs.  Note that I'm often warned about hangs when oversubscribing resources but many times vt recovers.

**Pull request merged**